File: peace/EAST/Java/src/eSTAssembly/SixTuple.java
package eSTAssembly;

public class SixTuple {
	int curNode;	//index of the current node
	int leftNode;	//index of node on the left
	int lOvlLen;	//the overlap length with + or -, now it's -
	int lDis; 		//the distance with + or -, now it's -
	int rightNode;	//the index of node on the right
	int rOvlLen;	//the overlap length with + or -, now it's +
	int rDis;		//the distance with + or -, now it's +
	
	public SixTuple() {
	}
	
	public SixTuple(int c, int i1, int i2, int i3, int i4, int i5, int i6) {
		curNode = c;
		leftNode = i1;
		lOvlLen = i2;
		lDis = i3;
		rightNode = i4;
		rOvlLen = i5;
		rDis = i6;
	}
	
	public SixTuple(int c) {
		curNode = c;
	}
	
	public void setSixTuple(int i1, int i2, int i3, int i4, int i5, int i6) {
		leftNode = i1;
		lOvlLen = i2;
		lDis = i3;
		rightNode = i4;
		rOvlLen = i5;
		rDis = i6;
	}
}

File: peace/EAST/C++/SixTuple.cpp
#include "SixTuple.h"

SixTuple::SixTuple() {
	this->curNode = 0;
	this->leftNode = 0;
	this->lOvlLen = 0;
	this->lDis = 0;
	this->rightNode = 0;
	this->rOvlLen = 0;
	this->rDis = 0;
	this->isNull = true;
}

SixTuple::SixTuple(int c, int i1, int i2, int i3, int i4, int i5, int i6, bool b) {
	this->curNode = c;
	this->leftNode = i1;
	this->lOvlLen = i2;
	this->lDis = i3;
	this->rightNode = i4;
	this->rOvlLen = i5;
	this->rDis = i6;
	this->isNull = b;
}

SixTuple::SixTuple(int c) {
	this->curNode = c;
	this->isNull = true;
	this->leftNode = 0;
	this->lOvlLen = 0;
	this->lDis = 0;
	this->rightNode = 0;
	this->rOvlLen = 0;
	this->rDis = 0;
}

void SixTuple::setSixTuple(int i1, int i2, int i3, int i4, int i5, int i6, bool b) {
	this->leftNode = i1;
	this->lOvlLen = i2;
	this->lDis = i3;
	this->rightNode = i4;
	this->rOvlLen = i5;
	this->rDis = i6;
	this->isNull = b;
}
